nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,386 @@
+                    cmap=cm.cold_hot, black_bg=False, symmetric_cmap=True,
+                    vol_to_surf_kwargs={}):
+    """Project 3d map on cortex; prepare info to plot both hemispheres."""
+    info = {}
+    mesh = _check_mesh(mesh)
+    surface_maps = {
+        h: surface.vol_to_surf(volume_img, mesh['pial_{}'.format(h)],
+                               **vol_to_surf_kwargs)
+        for h in ['left', 'right']
+    }
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, np.asarray(list(surface_maps.values())).ravel(), threshold,
+        symmetric_cmap=symmetric_cmap)
+
+    for hemi, surf_map in surface_maps.items():
+        bg_map = surface.load_surf_data(mesh['sulc_{}'.format(hemi)])
+        info['pial_{}'.format(hemi)] = to_plotly(mesh['pial_{}'.format(hemi)])
+        info['inflated_{}'.format(hemi)] = to_plotly(
+            mesh['infl_{}'.format(hemi)])
+
+        info['vertexcolor_{}'.format(hemi)] = _get_vertexcolor(
+            surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = True
+    info['colorscale'] = colors
+    return info
+
+
+def _fill_html_template(info, embed_js=True):
+    as_json = json.dumps(info)
+    as_html = _get_html_template().replace(
+        'INSERT_STAT_MAP_JSON_HERE', as_json)
+    as_html = add_js_lib(as_html, embed_js=embed_js)
+    return SurfaceView(as_html)
+
+
+def view_img_on_surf(volume_img, mesh='fsaverage5',
+                     threshold=None, cmap=cm.cold_hot,
+                     black_bg=False, embed_js=True):
+    """
+    Insert a surface plot of a statistical map into an HTML page.
+
+    Parameters
+    ----------
+    volume_img : Niimg-like object, 3d
+        See http://nilearn.github.io/manipulating_images/input_output.html
+
+    mesh : str or dict, optional.
+        if 'fsaverage5', use fsaverage5 mesh from nilearn.datasets
+        if 'fsaverage', use fsaverage mesh from nilearn.datasets
+        if a dictionary, keys should be 'infl_left', 'pial_left', 'sulc_left',
+        'infl_right', 'pial_right', 'sulc_right', containing inflated and
+        pial meshes, and sulcal depth values for left and right hemispheres.
+
+    threshold : str, number or None, optional (default=None)
+        If None, no thresholding.
+        If it is a number only values of amplitude greater
+        than threshold will be shown.
+        If it is a string it must finish with a percent sign,
+        e.g. "25.3%", and only values of amplitude above the
+        given percentile will be shown.
+
+    cmap : str or matplotlib colormap, optional
+
+    black_bg : bool, optional (default=False)
+
+    embed_js : bool, optional (default=True)
+        if True, jquery and plotly are embedded in resulting page.
+        otherwise, they are loaded via CDNs.
+
+    Returns
+    -------
+    SurfaceView : plot of the stat map.
```

Comment:
May be we should add the interesting methods to look at for users from the ```SurfaceView``` object. For me, the interesting things to showcase are ```open_in_browser```, ```save_as_html```, ```remove_temp_file```.

Right now, I see other ```get_iframe```, ```html``` and controlling ```height``` and ```width``` is for more advanced users IMO.

Reply:
I agree. I think `resize` can also be useful in a jupyter notebook, as in 
```
plotting.view_img_on_surf(img).resize(900, 600)
```
wich will display a bigger plot than the default. so that users can adjust the plot to what they want to fit on their screen, e.g. if they are giving a course and they want to show the cell and the plot at the same time